test/utils/Strings.test.js

```diff
@@ -120,6 +120,18 @@ describe('Strings', function () {
     });
   });
 
+  describe('toChecksumHexString address', function () {
+    it('converts a random address', async function () {
+      const addr = '0xa9036907dccae6a1e0033479b12e837e5cf5a02f';
+      expect(await this.mock.getFunction('$toChecksumHexString(address)')(addr)).to.equal(ethers.getAddress(addr));
+    });
+
+    it('converts an address with leading zeros', async function () {
+      const addr = '0x0000e0ca771e21bd00057f54a68c30d400000000';
+      expect(await this.mock.getFunction('$toChecksumHexString(address)')(addr)).to.equal(ethers.getAddress(addr));
+    });
+  });
```

Comment:
There's a little consideration in the [documentation of `getAddress`](https://docs.ethers.org/v6/api/address/#getAddress):

> *  If %%address%% contains both upper-case and lower-case, it is
> *  assumed to already be a checksum address and its checksum is
> *  validated, and if the address fails its expected checksum an
> *  error is thrown.

None of these tests are using mixed-case letters so I'd recommend adding `.toLowerCase()` according to the same docs:

> * If you wish the checksum of %%address%% to be ignore, it should
> *  be converted to lower-case (i.e. ``.toLowercase()``) before
> *  being passed in.

Even better, let's rewrite these tests:

```
const addresses = [...]

describe('toChecksumHexString address', function () {
  for (const addr of addresses) {
    it(`converts ${addr}`, async function () {
      expect(await this.mock.getFunction('$toChecksumHexString(address)')(addr.toLowerCase())).to.equal(
        ethers.getAddress(addr),
      );
    });
  }
});
```

I'm pushing a commit